android/src/play/java/com/dooboolab/RNIap/RNIapModule.kt

```diff
@@ -90,9 +92,17 @@ class RNIapModule(
             return
         }
 
-        billingClient = builder.setListener(this).build()
+        if (billingClientCache?.isReady == true) {
+            Log.i(
+                TAG,
+                "Already initialized, you should only call initConnection() once when your app starts"
+            )
+            promise.safeResolve(true)
+            return
+        }
+        billingClientCache = builder.setListener(this).build()
```

Comment:
@andresesfm I think we can help the garbage collector here too.
```
billingClientCache = null
billingClientCache = builder.setListener(this).build()
```

Reply:
That extra statement does not make a difference. The old reference will be garbage collected regardless 🙁